subsys/mpsl/fem/mpsl_fem.c

```diff
@@ -281,6 +380,48 @@ static int fem_nrf21540_gpio_configure(void)
 	}
 #endif
 
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO_SPI)
+#ifdef CONFIG_PINCTRL
+
+#define FEM_SPI_PIN_NUM(node_id, prop, idx) \
+	NRF_GET_PIN(DT_PROP_BY_IDX(node_id, prop, idx)),
+
+#define FEM_SPI_PIN_FUNC(node_id, prop, idx) \
+	NRF_GET_FUN(DT_PROP_BY_IDX(node_id, prop, idx)),
+
+    uint8_t fem_spi_pin_nums[] = {
+		DT_FOREACH_CHILD_VARGS(
+			DT_PINCTRL_BY_NAME(DT_BUS(MPSL_FEM_SPI_IF), default, 0),
+			DT_FOREACH_PROP_ELEM, psels, FEM_SPI_PIN_NUM
+		)
+	};
+
+    uint8_t fem_spi_pin_funcs[] = {
+		DT_FOREACH_CHILD_VARGS(
+			DT_PINCTRL_BY_NAME(DT_BUS(MPSL_FEM_SPI_IF), default, 0),
+			DT_FOREACH_PROP_ELEM, psels, FEM_SPI_PIN_FUNC
+		)
+	};
+
+	for (size_t i = 0U; i < ARRAY_SIZE(fem_spi_pin_nums); i++) {
+		switch (fem_spi_pin_funcs[i]) {
+			case NRF_FUN_SPIM_SCK:
+                pin_num_to_mpsl_fem_pin(fem_spi_pin_nums[i], &cfg.spi_config.sck_pin);
+				break;
+			case NRF_FUN_SPIM_MOSI:
+                pin_num_to_mpsl_fem_pin(fem_spi_pin_nums[i], &cfg.spi_config.mosi_pin);
+				break;
+			case NRF_FUN_SPIM_MISO:
+                pin_num_to_mpsl_fem_pin(fem_spi_pin_nums[i], &cfg.spi_config.miso_pin);
+				break;
+            default:
+				break;
+		}
+	}
+#endif
```

Comment:
Please add comments to determine which `#if` is terminated

Reply:
Fixed